src/components/fields/types.ts

```diff
@@ -0,0 +1,4 @@
+export interface IOption {
```

Comment:
understand the rationale of putting it here but it feels a bit awkward, in order to keep the folder structure clean, I think we can declare this interface in each field instead